In [ ]:
from google.colab import drive
import pandas as pd
import numpy as np
from google.colab import drive
drive.mount('/content/drive')
file_path='/content/drive/My Drive/datasets/binary_classification_train.csv'
data=pd.read_csv(file_path)
matrix = data.iloc[:, 1:].values

Mounted at /content/drive


In [ ]:
for k in range(5,15):
  X_train= np.array(matrix[:32000, :20])
  y_train= np.array(matrix[:32000, 20])
  X_test= np.array(matrix[32000:, :20])
  y_test=np.array(matrix[32000:, 20])
  prediction = np.zeros(len(X_test))
  i = 0
  c = 0
  for x in X_test:
      dists = np.linalg.norm(X_train - x, axis=1)
      idxs = np.array(sorted(range(len(dists)), key=lambda i: dists[i])[:k])
      labels = np.array([y_train[i] for i in idxs])
      counts = {}
      for l in labels:
          counts[l]=counts.get(l, 0) + 1
      prediction[i]=max(counts, key=counts.get)

      if prediction[i]==y_test[i]:
          c+=1
      i+=1

  accuracy = c/len(X_test)
  print(k)
  print(accuracy)

5
0.9470625
6
0.9473125
7
0.948625
8
0.9480625
9
0.9488125
10
0.94875
11
0.949125
12
0.949125
13
0.949
14
0.949


4: 94.15,
5:
0.9470625,
6:
0.9473125,
7:
0.948625,
8:
0.9480625,
9:
0.9488125,
10:
0.94875,
11:
0.949125,
12:
0.949125,
13:
0.949,
14:
0.949,
15:
0.948875,





On changing value of K, as well as selecting different subsets of the dataset, the accuracy remains around 92%-93% with K>2, and 10k training and test data.

with 32k training data and 16k test data, accuracy goes upto 94.9% at k>10

In [ ]:
#predictions
drive.mount('/content/drive')
file_path='/content/drive/My Drive/datasets/binary_classification_test.csv'
data=pd.read_csv(file_path)
matrix1 = data.iloc[:, 1:].values
X_train= np.array(matrix[:, :20])
y_train= np.array(matrix[:, 20])
X_test= np.array(matrix[:, :20])

prediction = np.zeros(len(X_test))
i = 0
c = 0
for x in X_test:
  dists = np.linalg.norm(X_train - x, axis=1)
  idxs = np.array(sorted(range(len(dists)), key=lambda i: dists[i])[:k])
  labels = np.array([y_train[i] for i in idxs])
  counts = {}
  for l in labels:
    counts[l]=counts.get(l, 0) + 1
    prediction[i]=max(counts, key=counts.get)



